lib/auth.js

```diff
@@ -126,6 +130,74 @@ var auth = (function () {
         }
     };
 
+    var checkOrganizations = function(req, res, next){
+	
+	if (!req.user || tokensCache[req.user.accessToken].orgProcessed || tokensCache[req.user.accessToken].orgProcessing){
+	    next();
+	} else {
+	    tokensCache[req.user.accessToken].orgProcessing = true;
+	    async.waterfall([ // here should be a list of functions
+		function(callback){
+		    party.getOrganizations((err, resp) => {
```

Comment:
This way of working is quite inefficient. Instead of retrieving all the organizations and then iterate over the req.user.organization field, iterate over it and then ask only for the specific organization using its id. If you receive a 404 means that you have to create it.